5.traits.md

```diff
@@ -72,9 +72,26 @@ The specification defines two major things: The list of workload types to which
 
 | Attribute | Type | Required | Default Value | Description |
 |-----------|------|----------|---------------|-------------|
+| `group` | `string` | Y | | The group that this trait belongs to (e.g. `core.oam.dev`)
+| `version` | `string` | Y | | The version of the trait API (e.g. `v1alpha1`)
+| `names` | [`Names`](#Names) | Y | | Information about the names associated with this trait |
 | `appliesTo` | `[]string` | Y | `["*"]` | The list of workload types that this trait applies to. `"*"` means _any workload type_. A trait must apply to at least one workload type. This attribute must contain at least one value. An empty array is invalid for this attribute. |
 | `properties` | [`Properties`](#properties) | Y | | The trait's configuration options. The value is a [JSON schema](https://json-schema.org/) expressed as json string |
 
+### Names
+
+This section describes the different naming references for this kind.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `kind` | `string` | Y | | The proper reference name of the trait (e.g., `AutoScaler`) |
+| `singular` | `string` | N | | The singular human-readable name of this type (e.g., `server`) |
+| `plural` | `string` | N | | The plural human-readable name of this type (e.g., `servers`) |
+
+`kind` must begin with a capital letter, and contain only characters 0030-0039, 0041-005a, and 0061-007a from the Unicode basic Latin characters.
```

Comment:
Is this a project decision to use Unicode basic Latin. Why not just UTF-8?

http://utf8everywhere.org/

Reply:
I just align it with Workload Type https://github.com/oam-dev/spec/blob/master/7.workload_types.md#names